MarkBind/markbind
# Fix external styles and script not hoisted

**What is the purpose of this pull request?**


Resolve issue: #2300 

**Overview of changes:**
Widen the scope of styles and script hoisting to include external and external manager.

**Anything you'd like to highlight/discuss:**

**Testing instructions:**
Init a markbind sample project.
In contents/topic1.md, add:
```
<h1>
<script>
      // JavaScript code specific to this component
      console.log("Inline script executed! 35");
</script>
<style>
      /* CSS styles specific to this component */
      h1 {
        color: red;
      }
</style>
h1 text
</h1>
```

In index.md, add
```
<panel src = "contents/topic1.md">
</panel>
```
You should see the following.The styles in the contents/topic1.md is applied to the content in the panel successfully.
<img width="689" alt="Screenshot 2024-02-12 at 15 02 34" src="https://github.com/MarkBind/markbind/assets/121547057/af83805b-4093-41d8-aef9-7e5310c53c3e">

**Proposed commit message: (wrap lines at 72 characters)**

Hoist external's styles and script
Also add warning in documentation that styles in hoisted
external will also affect the rest of the page
---

**Checklist:** :ballot_box_with_check:






---

**Reviewer checklist:**

Indicate the [SEMVER](https://semver.org/) impact of the PR:

At the end of the review, please label the PR with the appropriate label: `r.Major`, `r.Minor`, `r.Patch`.

Breaking change release note preparation (if applicable):
- To be included in the release note for any feature that is made obsolete/breaking

> Give a brief explanation note about:
>   - what was the old feature that was made obsolete
>   - any replacement feature (if any), and
>   - how the author should modify his website to migrate from the old feature to the replacement feature (if possible).
